go/app/main.go

```diff
@@ -28,14 +43,113 @@ func root(c echo.Context) error {
 func addItem(c echo.Context) error {
 	// Get form data
 	name := c.FormValue("name")
+	category := c.FormValue("category")
+	image := c.FormValue("image")
+
 	c.Logger().Infof("Receive item: %s", name)
+	c.Logger().Infof("Receive category: %s", category)
+	c.Logger().Infof("Receive image: %s", image)
+
+	hash := sha256.New()
+	// Write the image data to the hash function
+	hash.Write([]byte(strings.Split(image, ".")[0]))
+
+	// Get the final hash value
+	hashValue := hash.Sum(nil)
+	// Convert the byte slice to a hex-encoded string
+	hashString := hex.EncodeToString(hashValue)
+	image = hashString + ".jpg"
+
+	var res Response
+	var itemslice []Item
+	// Create a new item with the next ID
+	newItem := Item{
+		Name:     name,
+		Category: category,
+		Image:    image,
+	}
+
+	if _, err := os.Stat("items.json"); err == nil {
+		file1, err := os.Open("items.json") //すでにあるファイルを開く
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		defer file1.Close()
+		jsonData, err := ioutil.ReadAll(file1)
+		if err != nil {
+			fmt.Println("JSONデータを読み込めません", err)
+		}
 
-	message := fmt.Sprintf("item received: %s", name)
-	res := Response{Message: message}
+		json.Unmarshal(jsonData, &itemslice)
+
+		//Print message
+		message := fmt.Sprintf("item received: %s in %s category", newItem.Name, newItem.Category)
+		res = Response{Message: message}
+
+		// Append the new item to the slice
+		itemslice = append(itemslice, newItem)
+
+	} else {
+		itemslice = append(itemslice, newItem)
+
+	}
+
+	file2, err := os.Create("items.json") // fileはos.File型
+	if err != nil {
+		log.Fatal(err)
+	}
+	json.NewEncoder(file2).Encode(itemslice)
 
 	return c.JSON(http.StatusOK, res)
 }
 
+func getItems(c echo.Context) error {
+	file1, err := os.Open("items.json") //すでにあるファイルを開く
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer file1.Close()
+	jsonData, err := ioutil.ReadAll(file1)
+	if err != nil {
+		fmt.Println("JSONデータを読み込めません", err)
+	}
+	var itemslice []Item
+	json.Unmarshal(jsonData, &itemslice)
+	fmt.Println(itemslice)
+
+	return c.JSON(http.StatusOK, itemslice)
+}
+
+func getItem(c echo.Context) error {
+	itemIDStr := c.Param("item_id")
+
+	itemID, err := strconv.Atoi(itemIDStr)
+	if err != nil {
+		// エラーが発生した場合の処理
+		fmt.Println("item_idを整数に変換できません:", err)
```

Comment:
Printing the error is a good idea 👍 . 

But we need to always handle errors properly on the server.

We can improve this by:

1. Use `c.Logger().Infof` instead of `fmt.Println`. This will help us later because where `fmt.Println` writes to is not configurable.
2. Return the error and stop processing.